### PR TITLE
Use Chainer v2

### PR DESCRIPTION
**Pull request by toslunar**

### CODE REVIEW

File: chainerrl/agent.py [hunk 1]
```diff
@@ -107,7 +119,7 @@ def save(self, dirname):
     def load(self, dirname):
         """Load internal states."""
         for attr in self.saved_attributes:
-            serializers.load_npz(
+            load_npz_no_strict(
```
Comment (muupan) [inline chainerrl/agent.py]: What is the reason behind this change? It's better to add comments if this is a fix for some problem.

File: chainerrl/agents/a3c.py [hunk 2]
```diff
@@ -226,7 +226,9 @@ def update(self, statevar):
             target_link=self.shared_model, source_link=self.model)
         # Update the globally shared model
         if self.process_idx == 0:
-            norm = self.optimizer.compute_grads_norm()
+            # norm = self.optimizer.compute_grads_norm()
```
Comment (muupan) [inline chainerrl/agents/a3c.py]: Remove commented out code.

File: chainerrl/agents/acer.py [hunk 3]
```diff
@@ -523,7 +523,9 @@ def update(self, t_start, t_stop, R, states, actions, rewards, values,
             target_link=self.shared_model, source_link=self.model)
         # Update the globally shared model
         if self.process_idx == 0:
-            norm = self.optimizer.compute_grads_norm()
+            # norm = self.optimizer.compute_grads_norm()
```
Comment (muupan) [inline chainerrl/agents/acer.py]: Remove commented out code.

File: chainerrl/agents/pcl.py [hunk 4]
```diff
@@ -248,7 +248,10 @@ def update(self, loss):
             copy_param.copy_grad(
                 target_link=self.shared_model, source_link=self.model)
             if self.process_idx == 0:
-                norm = self.optimizer.compute_grads_norm()
+                # norm = self.optimizer.compute_grads_norm()
```
Comment (muupan) [inline chainerrl/agents/pcl.py]: Remove commented out code.

File: chainerrl/agents/pcl.py [hunk 5]
```diff
@@ -137,7 +137,7 @@ def __init__(self, model, optimizer,
         self.pi_loss_coef = pi_loss_coef
         self.v_loss_coef = v_loss_coef
         self.rollout_len = rollout_len
-        self.batchsize = batchsize
+        self.batchsize = self.xp.int32(batchsize)
```
Comment (muupan) [inline chainerrl/agents/pcl.py]: I guess this is a fix for newint, but it's not clear from the code. Can you add a comment?

File: chainerrl/initializers/normal.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import chainer
+import numpy as np
+
+
+class LeCunNormal(chainer.initializers.HeNormal):
+    """sorry
+
+    """
```
Comment (muupan) [inline chainerrl/initializers/normal.py]: Improve the docstring.

File: chainerrl/misc/async.py [hunk 7]
```diff
@@ -14,6 +14,13 @@
 from chainerrl.misc import random_seed
 
 
+def ensure_initialized_update_rule(param):
+    u = param.update_rule
+    if u.state is None:
+        u._state = {}  # Sorry!
```
Comment (muupan) [inline chainerrl/misc/async.py]: Eraborate the comment.

File: .travis.yml [hunk 8]
```diff
@@ -6,7 +6,14 @@ python:
...
   - pip install --upgrade pip setuptools wheel
+  - sed -i.bak '/chainer>/d' requirements.txt
+  - sed -i.bak '/chainer>/d' setup.py
   - pip install -r requirements.txt --only-binary=numpy,scipy
+  - git clone https://github.com/pfnet/chainer.git
+  - cd chainer
+  - git checkout _v2
+  - python setup.py install
+  - cd ..
   - pip install jupyter
```
Comment (muupan) [inline .travis.yml]: Now that chainer v2 is released, can you undo these changes?

File: .travis.yml [hunk 9]
```diff
@@ -6,7 +6,14 @@ python:
...
 install:
   - pip install --upgrade pip setuptools wheel
+  - sed -i.bak '/chainer>/d' requirements.txt
+  - sed -i.bak '/chainer>/d' setup.py
   - pip install -r requirements.txt --only-binary=numpy,scipy
+  - git clone https://github.com/pfnet/chainer.git
+  - cd chainer
+  - git checkout _v2
+  - python setup.py install
+  - cd ..
```
Comment (muupan) [inline .travis.yml]: Now that chainer v2 is released, can you undo these changes?

File: tests/misc_tests/test_async.py [hunk 10]
```diff
@@ -67,42 +68,56 @@ def test_share_states(self):
         opt_a.setup(model)
         arrays = async.share_states_as_shared_arrays(opt_a)
         opt_b = optimizers.RMSprop()
-        opt_b.setup(model)
+        opt_b.setup(copy.deepcopy(model))
+        opt_b.update()  # sorry
```
Comment (muupan) [inline tests/misc_tests/test_async.py]: Elaborate the comment.

File: tests/misc_tests/test_async.py [hunk 11]
```diff
@@ -67,42 +68,56 @@ def test_share_states(self):
...
         opt_c = optimizers.RMSprop()
-        opt_c.setup(model)
+        opt_c.setup(copy.deepcopy(model))
+        opt_c.update()  # sorry
```
Comment (muupan) [inline tests/misc_tests/test_async.py]: Ditto.

File: tests/misc_tests/test_async.py [hunk 12]
```diff
@@ -67,42 +68,56 @@ def test_share_states(self):
         opt_a.setup(model)
         arrays = async.share_states_as_shared_arrays(opt_a)
         opt_b = optimizers.RMSprop()
-        opt_b.setup(model)
+        opt_b.setup(copy.deepcopy(model))
```
Comment (muupan) [inline tests/misc_tests/test_async.py]: Why is copy.deepcopy required here?

**Comment by muupan** [comment]
Great work! Have you checked tests (including gpu and slow tests) in your environment?

**Comment by toslunar** [comment]
Some slow tests fail because of BLAS errors that seem to be chainer/chainer#2744.
I'm doing gpu tests.

**Comment by muupan** [comment]
LGTM
